File: bot/src/main/java/com/ryderbelserion/discord/bot/guilds/GuildListener.java
package com.ryderbelserion.discord.bot.guilds;

import com.ryderbelserion.discord.bot.Beidou;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.events.guild.GuildJoinEvent;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.jetbrains.annotations.NotNull;

public class GuildListener extends ListenerAdapter {

    private final Beidou beidou;

    public GuildListener(@NotNull final Beidou beidou) {
        this.beidou = beidou;
    }

    @Override
    public void onGuildJoin(@NotNull GuildJoinEvent event) {
        final Guild guild = event.getGuild();

        this.beidou.onGuildReady(guild);
    }

    @Override
    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
        /*final Guild guild = event.getGuild();
        final String id = guild.getId();

        if (!id.equalsIgnoreCase("1142362341095251998")) return;

        this.beidou.getGuildManager().getGuild(id).ifPresent(action -> {
            action.getConfig().test(guild, event.getAuthor());
        });*/
    }
}

File: src/main/kotlin/com/ryderbelserion/bot/listeners/core/GuildListener.kt
package com.ryderbelserion.bot.listeners.core

import com.ryderbelserion.bot.Beidou
import net.dv8tion.jda.api.events.guild.GuildJoinEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class GuildListener(private val plugin: Beidou) : ListenerAdapter() {

    override fun onGuildJoin(event: GuildJoinEvent) {
        // Create folder if it does not exist on guild join.
        this.plugin.createGuildFolder(event.guild)
    }
}